service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateAzureVmStep.java

```diff
@@ -0,0 +1,250 @@
+
+      computeManager
+          .virtualMachines()
+          .define(resource.getVmName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          // TODO: network, use a real network
+          .withExistingPrimaryNetwork(network)
+          .withSubnet(subnetName)
+          .withPrimaryPrivateIPAddressDynamic()
+          .withExistingPrimaryPublicIPAddress(existingAzureIp)
+          // See here for difference between 'specialized' and 'general' LinuxCustomImage, the
+          // managed disk storage option being the key factor
+          // https://docs.microsoft.com/en-us/azure/virtual-machines/linux/imaging#generalized-and-specialized
+          .withSpecializedLinuxCustomImage(azureConfig.getCustomDockerImageId())
+          .withExistingDataDisk(existingAzureDisk)
+          .withTag("workspaceId", resource.getWorkspaceId().toString())
+          .withTag("resourceId", resource.getResourceId().toString())
+          .withSize(VirtualMachineSizeTypes.STANDARD_D2S_V3)
+          .create(
+              Defaults.buildContext(
+                  CreateVirtualMachineRequestData.builder()
+                      .setName(resource.getVmName())
+                      .setRegion(Region.fromName(resource.getRegion()))
+                      .setResourceGroupName(azureCloudContext.getAzureResourceGroupId())
+                      .setNetwork(network)
+                      .setSubnetName(subnetName)
+                      .setDisk(existingAzureDisk)
+                      .setPublicIpAddress(existingAzureIp)
+                      .setImage(azureConfig.getCustomDockerImageId())
+                      .build()));
+    } catch (ManagementException e) {
+      // Stairway steps may run multiple times, so we may already have created this resource. In all
+      // other cases, surface the exception and attempt to retry.
+      // Azure error codes can be found here:
+      // https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors
+      if (StringUtils.equals(e.getValue().getCode(), "Conflict")) {
+        logger.info(
+            "Azure Vm {} in managed resource group {} already exists",
+            resource.getVmName(),
+            azureCloudContext.getAzureResourceGroupId());
+        return StepResult.getStepResultSuccess();
+      }
+      if (StringUtils.equals(e.getValue().getCode(), "ResourceNotFound")) {
+        logger.info(
+            "Either the disk, ip, or network passed into this createVm does not exist "
+                + String.format(
+                    "\nResource Group: %s\n\tIp Name: %s\n\tNetwork Name: %s\n\tDisk Name: %s",
+                    azureCloudContext.getAzureResourceGroupId(),
+                    ipResource.getIpName(),
+                    "TODO",
+                    diskResource.getDiskName()));
+      }
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
```

Comment:
Should this return `STEP_RESULT_FAILURE_FATAL` in the `ResourceNotFound` case, and return `STEP_RESULT_FAILURE_RETRY` otherwise?